商品削除機能

aki800 opened the pull request:
#WHAT
商品削除機能を実装

#WHY
商品を削除できるようにするため

商品を削除する様子↓（ビュー崩れが少しありますm(_ _)m）
https://gyazo.com/4f86eaa0febe4961658985ad80d4318a

Commit:
商品削除機能

#WHAT
商品削除機能を実装

#WHY
商品を削除できるようにするため

aki800 committed:
Merge branch 'master' into item-edit-destroy

ato-tech commented on app/assets/stylesheets/item_exhibitindex.scss:
```diff
@@ -0,0 +1,136 @@
+    display: block;
+  }
+
+  &_interval {
```
この場合BEM記法では__（ハイフン二つ）になります。細かいですが、全体をBEMに修正してください。

ato-tech commented on app/assets/stylesheets/item_exhibitindex.scss:
```diff
@@ -0,0 +1,136 @@
+      clear: both;
+
+      &_news, &_todo {
+        width: calc(700px / 3);
```
SASSは演算できますのでこの場合は700px / 3のみで大丈夫です。
calcを使うのは%とpxなど異なる単位の計算のみです。

ato-tech commented on app/controllers/items_controller.rb:
```diff
@@ -28,11 +28,23 @@ def show
   end
 
   def update
+    item = Item.find(params[:id])
```
編集機能はこのプルリクエストには含まれないため、削除しましょう。

ato-tech commented on app/controllers/items_controller.rb:
```diff
@@ -28,11 +28,23 @@ def show
+    end
+  end
+
+  def exhibitshow
```
こちらはなんでしょう？基本的にrailsの7つのアクションで実装するようにしましょう。

hiroki5123 commented on app/controllers/items_controller.rb:
```diff
@@ -28,11 +28,23 @@ def show
   def update
+    item = Item.find(params[:id])
+    item.update(item_params)
+    if item.update(item_params)
```
32行目と33行目で同じこと書いてませんか？

aki800 replied:
32行目を消去しました。

hiroki5123 commented on app/controllers/items_controller.rb:
```diff
@@ -28,11 +28,23 @@ def show
   def destroy
     item = Item.find(params[:id])
-    item.destroy if item.user_id == current_user.id
+    item.destroy
```
destroyできなかったときのエラーハンドリングを書きましょう

hiroki5123 commented on app/views/items/destroy.html.haml:
```diff
@@ -0,0 +1,50 @@
+        %li.my-page-main_notice_todo やることリスト
+      %ul.my-page-main_news-wrap
+        %li.my-page-main_news-list
+          %a{:href => "#"}
```
link_toで仮置きしておきましょう

aki800 replied:
全ての %a{:href => "#"}をlink_toで仮置きしました。

hiroki5123 commented on app/views/items/exhibitshow.html.haml:
```diff
@@ -0,0 +1,98 @@
+        %li.img{style: "background:#cfffcf;"}
+      #bx-pager-1
+        %li.sub-img{style: "background:#ffffcf;"}
+          %a{"data-slide-index" => "0", href: ""}
```
こちらも

aki800 replied:
link_toで仮置きしました。

hiroki5123 commented on app/views/users/exhibitindex.html.haml:
```diff
@@ -0,0 +1,35 @@
+        %li.my-page-main-exhibit_notice-index_todo 取引中
+        %li.my-page-main-exhibit_notice-index_todo 売却済み
+
+      - @items.each do |item|
```
renderのコレクションオプションを使いましょう

ghost commented:
そのビュー崩れはいつ修正するんでしょうか？

→商品詳細ページのビュー担当者の方に明日修正して頂きます。

ato-tech commented on app/views/items/destroy.html.haml:
```diff
@@ -0,0 +1,50 @@
+= render :partial => "layouts/main-header"
```
ハッシュロケットはやめましょう。

ato-tech commented on app/views/items/destroy.html.haml:
```diff
@@ -0,0 +1,50 @@
+        %li.my-page-main_notice_todo やることリスト
+      %ul.my-page-main_news-wrap
+        %li.my-page-main_news-list
+          %a{:href => "#"}
```
リンク先は仮置きでかまいませんので、link_toを使いましょう。

Commit:
削除機能修正

kuroko918 commented on app/controllers/items_controller.rb:
```diff
@@ -32,7 +32,11 @@ def update
-    item.destroy if item.user_id == current_user.id
+    if item.destroy
+      redirect_to controller: :users, action: :index
+    else item.destroy
```
rubyでは、elseに対して条件式を付与することはできませんし、ifと重複しているのですが…

Commit:
items_controllerの修正

#WHAT
Destroyの条件分岐を修正

#WHY
書き方を間違えていたため

aki800 commented:
rubyでは、elseに対して条件式を付与することはできませんし、ifと重複しているのですが…
↓
items_controllerのdestroyアクションの条件分岐の記述を修正しました。
(メンターさんのコメントを誤って消してしまいました。すみません。。）

miyagit commented on app/controllers/items_controller.rb:
```diff
@@ -28,13 +28,19 @@ def show
   def destroy
     item = Item.find(params[:id])
     if item.destroy
       redirect_to controller: :users, action: :index
```
redirect_to users_pathですかね？？

Commit:
items_controllerの修正2

#WHAT
リダイレクト先のパスの書き方を変更

#WHY
挙動を良くするため
よりコンパクトにするため

aki800 commented:
items_controllerのdestroyアクションの、リダイレクト先のパスの書き方を変更しました。

kuroko918 commented on app/controllers/items_controller.rb:
```diff
@@ -28,11 +28,21 @@ def show
   def update
+    item = Item.find(params[:id])
+    if item.update(item_params)
+      redirect_to controller: :items, action: :index
```
こちらもprefixを使用して記述しましょう！

Commit:
items_controllerの修正3

#WHAT
Updateのリダイレクト先のパスの書き方をprefixにしました。
しかしUpdateは、削除機能に入らないのでここでは消しました。お手数おかけします。

#WHY
効率の良いパスを記述するため。

kuroko918 commented:
一旦いいかと思います！
マージして次へ進んでください！
